twill-yarn/src/main/java/org/apache/twill/internal/appmaster/RunningContainers.java

```diff
@@ -267,6 +280,14 @@ void stopByIdAndWait(String runnableName, int instanceId) {
 
         resourceReport.removeRunnableResources(runnableName, containerId);
         containerChange.signalAll();
+        if (eventHandler != null) {
+          Integer exitStatus = containerExitStatus.get(containerId);
```

Comment:
Should remove it from the exit status map, otherwise the map can keep growing in size.